include/eve/algo/container/soa_vector2.hpp

```diff
@@ -0,0 +1,371 @@
+    EVE_FORCEINLINE auto cbegin_aligned() const -> const_iterator_aligned { return begin_aligned(); }
+
+    //! Returns an iterator to the beginning
+    EVE_FORCEINLINE auto begin() -> iterator { return views::convert(data(), eve::as<Type>{}); }
+
+    //! Returns an iterator to the beginning
+    EVE_FORCEINLINE auto begin()  const -> const_iterator { return views::convert(data(), eve::as<Type>{}); }
+
+    //! Returns a constant iterator to the beginning
+    EVE_FORCEINLINE auto cbegin() const -> const_iterator { return begin(); }
+
+    //! Returns an iterator to the end
+    EVE_FORCEINLINE auto end()  -> iterator  { return begin() + size(); }
+
+    //! Returns an iterator to the end
+    EVE_FORCEINLINE auto end()  const -> const_iterator { return begin() + size(); }
+
+    //! Returns a constant iterator to the end
+    EVE_FORCEINLINE auto cend() const -> const_iterator { return end(); }
+
+    //==============================================================================================
+    //! @}
+    //==============================================================================================
+
+    //==============================================================================================
+    //! @name Comparisons and ordering
+    //! @{
+    //==============================================================================================
+    friend bool operator==(soa_vector const& lhs, soa_vector const& rhs)
+    {
+      if( lhs.size() != rhs.size() ) return false;
+      return eve::algo::equal(lhs, rhs.begin_aligned());
+    }
+
+    friend bool operator!=(soa_vector const& lhs, soa_vector const& rhs)
+    {
+      return !(lhs == rhs);
+    }
+
+    //==============================================================================================
+    //! @}
+    //==============================================================================================
+
+    friend std::ostream& operator<<(std::ostream& os, soa_vector const& v)
+    {
+      os << "{ ";
+      for(std::size_t i=0;i<v.size();++i) os << v.get(i) << " ";
+      os << '}';
+      return os;
+    }
+
+    template <typename Traits, typename Self>
+    requires std::same_as<std::remove_reference_t<Self>, soa_vector>
+    EVE_FORCEINLINE friend auto tagged_dispatch(algo::preprocess_range_, Traits tr, Self& self)
+    {
+      return algo::preprocess_range(tr, algo::as_range(self.begin_aligned(), self.end()));
+    }
+
+    private:
+
+    void grow(std::size_t new_capacity, std::size_t old_size)
+    {
+      soa_vector that(no_init, new_capacity, get_allocator());
+      eve::algo::copy(*this, that.begin_aligned());
+      this->operator=(std::move(that));
+      size_ = old_size;
+    }
+
+    void fill(std::size_t first, std::size_t last, value_type v)
+    {
+      eve::algo::fill(algo::as_range(begin() + first, begin() + last), v );
+    }
+
+    void fill(value_type v) { eve::algo::fill(*this, v ); }
```

Comment:
maybe this needs to be public, sounds like a useful function member to have?

Reply:
I don't know: eve::algo::fill(*this, v ) is not too much to call.

array has it.  

I suspect that's not what I meant though.

https://en.cppreference.com/w/cpp/container/array/fill

Reply:
Ah nah it's fine.

If you want `eve::algo::fill(*this, v );` in public, you can. It will save users another version of `eve::algo::fill` compiled in the binary.

Why not fill up to capacity? That will save you cycles.

Reply:
isn't it what this fill on the whole range does anyway ?